uint32 data
bool shouldCheckForAgentID
uint32[] agentIDs